maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java

```diff
@@ -112,6 +114,9 @@ public void build( MavenSession session, ReactorContext reactorContext, ProjectB
             }
 
         }
+
+        executor.shutdown();
+        executor.awaitTermination( Long.MAX_VALUE, TimeUnit.MILLISECONDS  );
```

Comment:
MAXVALUE... what is the risk of hanging builds in CI? If any, lowering might be a good idea?